deepsubtree.go

```diff
@@ -0,0 +1,151 @@
+package iavl
```

Comment:
Please add documentation to this file. 

Method docstrings, inline comments to help with understanding. Maybe a code block at the top as an overview.

Reply:
Right now it's just a proof of concept, to show that we can build deep subtree without range proofs.
Definitely lot of work needed to convert from draft to PR.

Reply:
Updated documentation and fixed it so the tests work